frontend/src/App.tsx

```diff
@@ -0,0 +1,180 @@
+import React, {useState, useEffect, useMemo, FC, ComponentProps, MouseEventHandler} from "react";
+import {CustomModal as Modal} from "./components/Modal";
+import axios from "axios";
+import {TodoItem} from "./models/TodoItem";
+
+axios.defaults.xsrfCookieName = 'csrftoken'
+axios.defaults.xsrfHeaderName = 'X-CSRFToken'
+
+interface TabListProps {
+    viewCompleted: boolean;
+    displayCompleted: (status: boolean) => void;
+}
+
+const TabList: FC<TabListProps> = ({viewCompleted, displayCompleted}) => {
+  return (
+    <div className="nav nav-tabs">
+      <span
+        className={viewCompleted ? "nav-link active" : "nav-link"}
+        onClick={() => displayCompleted(true)}
+      >
+        Complete
+      </span>
+      <span
+        className={viewCompleted ? "nav-link" : "nav-link active"}
+        onClick={() => displayCompleted(false)}
+      >
+        Incomplete
+      </span>
+    </div>
+  );
+};
+
+interface ItemsProps {
+  todoList: TodoItem[];
+  viewCompleted: boolean;
+  handleDelete: (item: TodoItem) => void;
+  editItem: (item: TodoItem) => void;
+}
+
+const Items: FC<ItemsProps> = ({todoList, viewCompleted, handleDelete, editItem}) => {
+  const newItems = useMemo(() => todoList.filter(
+    (item) => item.completed === viewCompleted
+  ), [todoList, viewCompleted]);
+
+  return (
+    <>
+      {newItems.map((item) => (
+        <li
+          key={item.id}
+          className="list-group-item d-flex justify-content-between align-items-center"
+        >
+          <span
+            className={`todo-title mr-2 ${
+              viewCompleted ? "completed-todo" : ""
+            }`}
+            title={item.description}
+          >
+            {item.title}
+          </span>
+          <span>
+            <button
+              className="btn btn-secondary mr-2"
+              onClick={() => editItem(item)}
+            >
+              Edit
+            </button>
+            <button
+              className="btn btn-danger"
+              onClick={() => handleDelete(item)}
+            >
+              Delete
+            </button>
+          </span>
+        </li>
+      ))}
+    </>
+  );
+};
+
+export const App = () => {
+  const [viewCompleted, setViewCompleted] = useState(false);
```

Comment:
eventhough is visible the type, is a good idea to use the type in useState 
```js
const [viewCompleted, setViewCompleted] = useState<boolean>(false);
```

Reply:
fair enough